### PR TITLE
8606 - Clicking glossary terms on the COVID page opens the Glossary without refreshing the page

### PR DESCRIPTION
**Pull request by anjenkin**
https://federal-spending-transparency.atlassian.net/browse/DA-8606

Please note, this doesn't apply to the terms in the Spending Definitions modal.  That has been separated out into another ticket.

### CODE REVIEW

File: src/components/glossary/glossary.jsx [hunk 1]
```diff
@@ -320,32 +329,34 @@ export default function Glossary(props) {
...
-				if (termSelected && keyCode !== enterKey) {
+				const keyCode = ev && ev.key ? ev.key : '';
+				const evType = ev && ev.type ? ev.type : '';
+				if ((termSelected && keyCode !== enterKey) || evType === 'change') {
 					return;
 				}
+
 				createTermHTMLElements(null, searchStr);
-				setTermListView(showListResultsInd);
+				// setTermSelected(null);
```
Comment (brianpetway) [inline src/components/glossary/glossary.jsx:359]: can this be removed?

File: src/components/glossary/glossary.jsx [hunk 2]
```diff
@@ -426,33 +435,44 @@ export default function Glossary(props) {
...
 	}, []);
 
 	useEffect(() => {
-		if (glossaryWrapper) {
-			filteredData = glossaryData.filter(r => r.term); //remove blank rows
-
-			origCategorizedTerms = categorizeGlossaryData(filteredData);
-
+		if (glossaryWrapper && filteredData && origCategorizedTerms) {
+			console.log(glossaryWrapper);
```
Comment (brianpetway) [inline src/components/glossary/glossary.jsx:450]: console.log left in

**Comment by brianpetway** [comment]
Wow, that was a lot of work. Great idea to make a component for the glossary terms in the copy.
A couple of inline comments above, to remove a commented out line and a console.log.

But I noticed some strange behavior - if you load the page at desktop size, clicking on a copy term will open the glossary correctly, then shrinking to tablet size they also work correctly, but then if you shrink down to mobile, clicking on the copy terms does not open the glossary. You can then open the glossary from the menu option, and then the copy terms will open the glossary at mobile size - but only while the menu dropdown is visible. If you expand out to tablet the same thing happens - clicking the copy terms does not open the glossary until after you've opened the glossary form the menu option and left the menu dropdown active. And if you then expand out to desktop, clicking on the copy terms only works if the glossary is open.
So, essentially, it seems like there is some issue with resizing.

**Comment by anjenkin** [comment]
@brianpetway Fixed issue with glossary on resize however, I had to do a big refactor to clean up the menu code to get this to work.  Sorry for the big PR.

**Comment by brianpetway** [comment]
Build failed because of an import in about.jsx - StoryPageHeader - looks like the path is not valid

**Comment by brianpetway** [comment]
At desktop and tablet, the first time you click on a glossary term, the glossary animation doesn't happen, the glossary just pops in; but after the first click, the animation does work

**Comment by anjenkin** [comment]
@brianpetway Should be fixed now

**Comment by brianpetway** [comment]
The initial behavior on page load is fixed, but it looks like we no longer have underline on hover on the glossary terms now

**Pull request merged**